core/chaincode/exectransaction_test.go

```diff
@@ -192,6 +190,28 @@ func TestExecuteDeployTransaction(t *testing.T) {
 	closeListenerAndSleep(lis)
 }
 
+// Test deploy of a transaction
+func TestExecuteDeployTransaction(t *testing.T) {
+	executeDeployTransaction(t, "github.com/hyperledger/fabric/examples/chaincode/go/chaincode_example01")
+}
+
+// Test deploy of a transaction with a GOPATH with multiple elements
+func TestGopathExecuteDeployTransaction(t *testing.T) {
+	// add a trailing slash to GOPATH
+	// and a couple of elements - it doesn't matter what they are
+	os.Setenv("GOPATH", os.Getenv("GOPATH") + string(os.PathSeparator) + string(os.PathListSeparator) + "/tmp/foo" + string(os.PathListSeparator) + "/tmp/bar")
+	fmt.Printf("set GOPATH to: \"%s\"\n", os.Getenv("GOPATH"))
+	executeDeployTransaction(t, "github.com/hyperledger/fabric/examples/chaincode/go/chaincode_example01")
+}
+
+// Test deploy of a transaction with a chaincode over HTTP.
+func TestHTTPExecuteDeployTransaction(t *testing.T) {
+	// The chaincode used here cannot be from the hyperledger repo
+	// itself or it won't be downloaded because it will be found
+	// in GOPATH, which would defeat the test
+	executeDeployTransaction(t, "http://github.com/lehors/fabric/examples/chaincode/go/chaincode_example01")
```

Comment:
I understand. It seems that both solutions have their pros and cons but I agree repeatability is important, and arguably more so than protecting ourselves against possible future changes which could be dealt with as part of those changes.

I very much care about having such a test be part of CI - I don't care so much how we get it done. I just wouldn't have felt comfortable touching that code without a way to test and get reasonably confident that I didn't break anything.

Given the implications maybe the right thing to do is for me to take it out of this PR and raise a separate issue and PR.

And since we're into fine-tuning this now, I think we should create a repo dedicated to holding that code, something like hyperleger/fabric-test, that we could more reliably depend on rather that someone else's repo - be mine or any other.

Every now and then or when necessary one will need to update it and update the test accordingly.